### ISSUE TITLE
Generalize connection box lookahead from SymbiFlow

### ISSUE THREAD

**Issue by HackerFoo** [verilog-to-routing/vtr-verilog-to-routing#1146]
#### Proposed Behaviour
Generate and use a lookahead map based on entry and exit from routing fabric.

#### Current Behaviour
The existing map lookahead only considers the current wire type and direction of the target.

#### Possible Solution
We have already implemented this lookahead ("connection box lookahead") in a fork, but it may be specific to SymbiFlow. This should be generalized to work with existing benchmarks.

#### Context
The connection box lookahead has worked well for SymbiFlow, and so it makes sense to generalize this work rather than keep it in a fork.

Related to https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1117

**Comment by kmurray** [comment]
Yep this makes sense. I've recently generalized the mainline map lookahead to capture the SOURCE->OPIN->CHANX/Y components (i.e. routing fabric entry). It would make sense to also capture the routing fabric exit (i.e. CHANX/CHANY -> IPIN -> SINK).

To figure out the right path forward it would be good to understand what characteristics of the routing architectures you're working on that are important to capture in the lookahead.
